src/frontend/src/components/FillForms/UserDataForm.tsx

```diff
@@ -1,337 +1,294 @@
+    (state) => state.search.loadingExpungementPacket
+  );
 
-  phoneNumberPattern = new RegExp(".*[0-9].*");
-  zipCodePattern = new RegExp("[0-9][0-9][0-9][0-9][0-9].*");
+  const [showModal, setShowModal] = useState(false);
 
-  public validateForm = () => {
-    const phoneNumberMatch = this.phoneNumberPattern.exec(
-      this.state.phoneNumber
-    );
-    const zipCodeMatch = this.zipCodePattern.exec(this.state.zipCode);
-    return new Promise((resolve: (value?: unknown) => void) => {
-      this.setState<any>(
-        {
-          invalidZipCode: this.state.zipCode.length > 0 && !zipCodeMatch,
-          invalidPhone: this.state.phoneNumber.length > 0 && !phoneNumberMatch,
-          invalidBirthDate:
-            this.state.dob.length > 0 &&
-            moment(this.state.dob, "M/D/YYYY", true).isValid() === false,
-        },
+  function buildName() {
+    const { first_name, middle_name, last_name } = aliases[0];
+    let fullName = "";
+    fullName += first_name ? first_name + " " : "";
+    fullName += middle_name ? middle_name + " " : "";
+    fullName += last_name;
+
+    return fullName.includes("*") ? "" : fullName || " ";
+  }
+
+  function allInputsValid() {
+    const errorMessages = {
+      "Zip code must lead with five digits":
+        isPresent(zipCode) && !isZipCode(zipCode),
+      "Phone number must be a valid 10 digit number":
+        isPresent(phoneNumber) && !isPhoneNumber(phoneNumber),
+    };
 
-        resolve
-      );
-    });
-  };
+    if (errorMessagesObjectToArray(errorMessages).length === 0) return true;
+
+    setErrorMessages(errorMessagesObjectToArray(errorMessages));
+    return false;
+  }
 
-  public handleChange = (e: React.BaseSyntheticEvent) => {
-    // See https://github.com/DefinitelyTyped/DefinitelyTyped/issues/26635 for why we're
-    // using the "any" type.
-    this.setState<any>({
-      [e.target.id]: e.target.value,
-    });
-  };
+  function downloadPacket() {
+    dispatch(
+      downloadExpungementPacket(
+        name,
+        dob,
+        mailingAddress,
+        phoneNumber,
+        city,
+        state,
+        zipCode
+      )
+    );
+  }
 
-  public handleSubmit = (e: React.BaseSyntheticEvent) => {
+  function handleSubmit(e: React.BaseSyntheticEvent) {
     e.preventDefault();
-    this.validateForm().then(() => {
-      if (
-        !this.state.invalidZipCode &&
-        !this.state.invalidPhone &&
-        !this.state.invalidBirthDate
-      ) {
-        return this.props.downloadExpungementPacket(
-          this.state.name,
-          this.state.dob,
-          this.state.mailingAddress,
-          this.state.phoneNumber,
-          this.state.city,
-          this.state.state,
-          this.state.zipCode
-        );
-      }
-    });
-  };
 
-  public componentDidMount() {
-    if (!(this.props.aliases.length > 0)) {
-      history.push("/record-search");
-    }
+    if (!allInputsValid()) return;
+
+    setErrorMessages([]);
```

Comment:
I see that `setErrorMessages` is called in `allInputsValid`. Are you thinking we call it again here just to make it explicit that we are setting it to `[]`?

Reply:
The `setErrorMessages([]);` on line 85 is to clear the error messages after successful submission. Without it, error messages persist even if you successfully download the packet.